SwiftIOSExperiments/Modules/Login/Interactor/LoginInteractor.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  LoginInteractor.swift
+//  SwiftIOSExperiments
+//
+//  Created by James Santos on 15/2/17.
+//  Copyright © 2017 James Santos. All rights reserved.
+//
+
+import UIKit
+
+class LoginInteractor: LoginUseCase {
+
+	weak var output: LoginInteractorOutput!
+
+	func authenticateUser(_ auth: Auth) {
+		AuthService.authenticate(auth.email, auth.password, completion: { [weak self] result in
+			DispatchQueue.main.async {
+				switch result {
+				case .success(let credentials):
+					guard let idToken = credentials.accessToken else { return }
+
+					AuthService.token = idToken
+					self?.output.authenticationSuccess(idToken)
```

Comment:
don't use keychain if you don't need it. Before the kick off & session lifetime we can't save in persistant object uselessly. 

The subject of my comment is to keep all your service complexity in the service code, not the tool you can use.

In your authentication service protocol you don't need the token parameter.